kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -42,9 +39,323 @@ import cats.syntax.either._
 import com.snowplowanalytics.snowplow.collectors.scalastream.model._
 import com.snowplowanalytics.snowplow.collectors.scalastream.sinks.KinesisSink.SqsClientAndName
 
+/**
+  * Kinesis Sink for the Scala collector.
+  */
+class KinesisSink private (
+  client: AmazonKinesis,
+  kinesisConfig: Kinesis,
+  bufferConfig: BufferConfig,
+  streamName: String,
+  executorService: ScheduledExecutorService,
+  maybeSqs: Option[SqsClientAndName]
+) extends Sink {
+  import KinesisSink._
+
+  private val WithBuffer = maybeSqs.isDefined
+
+  // Records must not exceed MaxBytes - 1MB (for Kinesis)
+  // When SQS buffer is enabled MaxBytes has to be 256k,
+  // but we encode the message with Base64 for SQS, so the limit drops to 192k
+  private val SqsLimit     = 192000 // 256000 / 4 * 3
+  private val KinesisLimit = 1000000
+  override val MaxBytes    = if (WithBuffer) SqsLimit else KinesisLimit
+  // private val BackoffTime          = 3000L
+
+  private val ByteThreshold   = bufferConfig.byteLimit
+  private val RecordThreshold = bufferConfig.recordLimit
+  private val TimeThreshold   = bufferConfig.timeLimit
```

Comment:
I think it's all because of `MaxBytes` which is capitalized in the `Sink` trait. It sort of makes sense, since they are constants, but I see your point.